### ISSUE TITLE
Metrics: Add a `created at` key in the @metadata hash

### ISSUE THREAD

**Issue by ph** [elastic/logstash#3899]
Currently when we create an event we set a default `@timestamp` with `Time.now` we should also save this information inside the `@metadata` key `created_at` this would allow people to use conditionals and do special if the event was in the pipeline for too long.

**Comment by jsvd** [comment]
This should include a change in the way we add the `@timestamp`: it should be the last thing done before `queue << event`:

```ruby
event["@timestamp"] ||= event["[@metadata][created_at]"]
queue << event
```

**Comment by ph** [comment]
I don't think we actually need that kind of change, because the difference between setting the value inside the `initialize` and just before enqueue would be minimal and negligible?  

Also this would require to go around all plugins to implement it, I think the difference in number vs the simplicity of implementing at initialize doesn't worth it.

**Comment by ph** [comment]
I dont think this belongs in the metrics project, but I still think this is something nice to be added to the event.
